### ISSUE TITLE
rvierdiiev - Payment for liquidation is not fair

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-06-symmetrical-judging#97]
rvierdiiev

medium

# Payment for liquidation is not fair

## Summary
Payment for liquidation is not fair and do not incentivize liquidators to do full job.
## Vulnerability Detail
In order to liquidate partyA 4 steps needed.
1.You need to [call `liquidatePartyA`](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L20-L32), which is not very gas consumptive function.
2.[Call `setSymbolsPrice`](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L34C14-L97), which needs much more gas and depends on amount of symbols that partyA uses.
3.[Call `liquidatePendingPositionsPartyA`](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L99-L124) which gas amount depends on amount of pending positions.
4.[call `liquidatePositionsPartyA`](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L125-L238) which can consume big amount of gas if user has open positions.

The problem is that liquidation payment is paid in the last step and it's paid [for 2 liquidators](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L221-L222). This 2 liquidators were set in the [step1](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L31) and [step2](https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/liquidation/LiquidationFacetImpl.sol#L88).

Obviously step1 consumes less amount of gas then any other functions. That's why liquidation fee should not be paid equally.
Also step3 and step4 can be executed by anyone and liquidators will receive their fee. This actually doesn't incentivize them to do the job till the end. They can do their 1 and 2 step and then just wait. Maybe trader, that is liquidated will do that by himself just to be able to trade again.

But the bigger problem that i see here is that one liquidator will pay more gas than another one and as result will receive less fee.
## Impact
Liquidators are not incentivized to do the job from beginning to finish. They will receive their payment anyway, so they can wait on step 3 and 4. 2 liquidators receive same payment for different job done.
## Code Snippet
Provided above
## Tool used

Manual Review

## Recommendation
I think that the one he started liquidation should do all next steps. Also in case if he doesn't want to call next steps in some period, then his fee should be lost. In this way, once liquidator initiated liquidation, he will be interested to finish it. Otherwise he will lose fees and also gas that he spent on initiation. 

Duplicate of #231
